### PR TITLE
Add github action to conditionally update Algolia index

### PR DESCRIPTION
**Pull request by Dhaulagiri**
### :pushpin: Summary
This adds an action that will only run a) on pushes to `main`  and b) when website content has changed.

#### TODO
- [x] Add secrets to the action
- [x] Add Algolia secrets to GitHub so the action can access them

### :link: External links

Jira ticket: [HDS-2846](https://hashicorp.atlassian.net/browse/HDS-2846)

***

:speech_balloon: Please consider using [conventional comments](https://conventionalcomments.org/) when reviewing this PR.

[HDS-2846]: https://hashicorp.atlassian.net/browse/HDS-2846?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: .github/workflows/update-search-index.yml [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    name: Get files changed and conditionally skip updating the index
+    outputs:
+      trigger-ci: ${{ steps.read-files.outputs.trigger-ci }}
+    steps:
+      - uses: actions/checkout@3df4ab11eba7bda6032a0b82a6bb43b11571feac # v4.0.0
+        with:
+          fetch-depth: 0  
+      - name: Get changed files
+        id: read-files
+        run: ./.github/scripts/filter_changed_files.sh "website" "packages/flight-icons/catalog.json"
```
Comment (didoo) [inline .github/workflows/update-search-index.yml:23]: we should look also at the design tokens (they're indexed too)

Comment (Dhaulagiri) [reply .github/workflows/update-search-index.yml:23]: do you have the path for that? should we also be triggering website CI for similar changes?

Comment (Dhaulagiri) [reply .github/workflows/update-search-index.yml]: Updated

**Comment by Dhaulagiri** [comment]
We can't really test this until it's in `main`, but it'll be easier to iterate on any necessary fixes once this is there

**Review by didoo — APPROVED** [review]

**Pull request merged**